templates/class.mustache

```diff
@@ -36,30 +48,55 @@ export default class {{&className}} {
         this.errorHandlers.push(handler);
     }
 
-    private request(method: string, url: string, body: any, headers: any, queryParameters: any, form: any, reject: CallbackHandler, resolve: CallbackHandler) {
+    setRequestHeadersHandler(handler: RequestHeadersHandler) {
+        this.requestHeadersHandler = handler;
+    }
+
+    setConfigureAgentHandler(handler: ConfigureAgentHandler) {
+        this.configureAgentHandler = handler;
+    }
+
+    setConfigureRequestHandler(handler: ConfigureRequestHandler) {
+        this.configureRequestHandler = handler;
+    }
+
+    private request(method: string, url: string, body: any, headers: RequestHeaders, queryParameters: any, form: any, reject: CallbackHandler, resolve: CallbackHandler) {
         if(this.logger) {
           this.logger.log(`Call ${method} ${url}`);
         }
 
-        let req = (request as SuperAgentStatic)(method, url).query(queryParameters);
+        let agent = this.configureAgentHandler
```

Comment:
Not related to your changes, but realized this can be a const as its not reassigned. Want to fix this up at the same time? :)